src/test/cypress/support/requests/CourseManagementRequests.ts

```diff
@@ -57,11 +57,20 @@ export class CourseManagementRequests {
      * @param title the title of the programming exercise
      * @param programmingShortName the short name of the programming exercise
      * @param packageName the package name of the programming exercise
+     * @param skipGroupSynchronization Some tests already waited for group synchronization. In this case the check should be skipped
```

Comment:
Maybe i understand this wrong but did you not remove all the static waits for group synchronization from the tests? What is this parameter for?

Reply:
For performance. Basically what this PR adds is that every time we create a programming exercise we do the static wait for group synchronization (if the parameter is set to true in the config). But if the test previously already created a programming exercise for an existing course we don't have to wait for the group synchronization again.
This is the case for the `ProgrammingExercise.spec`. There we create a course in the `before` hook. The first test creates a programming exercise in the UI, so it has to wait for the group synchronization. The second test creates a programming exercise programmatically. There we need to set `skipGroupSynchronization` to `true` otherwise the test would wait for another 65 seconds in vain (because it uses the same course as the first test).
This parameter is currently only used in the `ProgrammingExercise.spec`. We could fix the issue there by simply changing the order of the tests. Then we could remove the parameter. I left it there because maybe a future test runs into the same problem (has to create multiple programming exercises in the same course).

Reply:
Makes a lot of sense. I consider this conversation resolved :)